app/modules/auth/auth.factory.js

```diff
@@ -6,18 +6,36 @@ require('./auth.module.js')
 var OAuth = require('panoptes-client/lib/oauth');
 
 // @ngInject
-function authFactory($interval, $timeout, $location, $window, localStorageService, ModalsFactory, zooAPI, zooAPIConfig, CribsheetFactory, $rootScope) {
+function authFactory($location, $rootScope, AnnotationsFactory, zooAPI, CribsheetFactory) {
 
     var factory;
 
     var _user = {};
 
     OAuth.checkCurrent()
-      .then(function (user) {
-        if (user) {
-            _setUserData();
-        }
-      });
+        .then(function (user) {
+            if (user) {
+                _setUserData();
+            }
+            return user;
+        })
+        .then(function(user) {
```

Comment:
`user` here will still be defined even after they've logged out, so you end up warning volunteers that they're about to be automatically logged-out, after they've logged out.

Reply:
Ah I see. I haven't managed to find a suitable place in this flow to clear the user object in order to prevent that bug. Do you have any thoughts?

Reply:
You should stop checking for a token after the user logs out, but I don't know where in the SW code the user gets unset after logging out. I think ASM are handling this by storing the current user, and checking the stored user in https://github.com/zooniverse/anti-slavery-manuscripts/pull/251

Reply:
I was checking the wrong user: `_user`, not `user`. Should be fixed now.